common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -388,6 +388,55 @@ public static CharSequence escapeCsv(CharSequence value) {
     }
 
     /**
+     * Unescapes the specified escaped CSV field, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     *
+     * @param value The escaped CSV field which will be unescaped according to
+     *              <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>
+     * @return {@link CharSequence} the unescaped value if necessary, or the value unchanged
+     */
+    public static CharSequence unescapeCsv(CharSequence value) {
+        int length = checkNotNull(value, "value").length();
+        if (length == 0) {
+            return value;
+        }
+        int last = length - 1;
+        boolean quoted = isDoubleQuote(value.charAt(0)) && isDoubleQuote(value.charAt(last)) && length != 1;
+        StringBuilder unescaped = InternalThreadLocalMap.get().stringBuilder();
+        for (int i = 0; i < length; i++) {
+            char current = value.charAt(i);
+            switch (current) {
+                case DOUBLE_QUOTE:
+                    if (i == 0 || i == last) {
+                        if (length == 1) {
+                            // value is "\""
+                            throw new IllegalArgumentException("Invalid escaped csv field: " + value);
+                        }
+                        continue;
+                    } else {
+                        if (quoted && isDoubleQuote(value.charAt(i + 1)) && (i + 1) != last) {
+                            // Must be enclosed with double-quotes and
+                            // followed by a double-quote (not the last character)
+                            i++;
```

Comment:
Doesn't really matter. Javac generates the same bytes for both `i++` and `++i` if the return value is not used.